### PR TITLE
Add ttlSecondsAfterFinished into LDAP sync cron job

### PR DESCRIPTION
**Pull request by sochotnicky**
Upstream Kubernetes documentation is here:
https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/

What this change does - it makes sure that failed jobs do not linger but are
cleaned up by OpenShift. In LDAP sync case the customer should only care about
last sync success or failure.

This should help with alert fatigue from KubeJobFailed when there is a glitch
with LDAP sync.

Similar (unrelated) change in CNO:
https://github.com/openshift/cluster-network-operator/pull/1318

Version(s): 4.8+

Issue:

Link to docs preview:

Additional information:

### CODE REVIEW

**Comment by sochotnicky** [comment]
@openshift/team-documentation 

I suppose this should not be too controversial of a change

**Review by bergerhoffer — COMMENTED** [review]
LGTM from doc side

**Comment by bergerhoffer** [comment]
@stlaz @xingxingxia Can you see if this update to the cron job config for auto LDAP syncing is okay?

Preview (single file): http://file.rdu.redhat.com/~ahoffer/2022/ldap-syncing.html#ldap-auto-syncing_ldap-syncing-groups

**Comment by sochotnicky** [comment]
Actually I think I should probably make one adjustment - add an additional note about the value of ttlSecondsAfterFinished. It needs to roughly match the period that the job is configured to run. So if the job runs every 30 minutes (as the example) it is 1800 (seconds). If the schedule is different this needs to be adjusted as well

**Comment by bergerhoffer** [comment]
@sochotnicky You could do this by adding a callout to that line, like:

**Comment by bergerhoffer** [comment]
Also, after you make that update, please be sure to squash your commits to a single commit. Thanks!

**Comment by sochotnicky** [comment]
I believe the note that I added could use some word-smithing but generally the content should be OK.

File: modules/ldap-auto-syncing.adoc [hunk 1]
```diff
@@ -209,9 +210,10 @@ spec:
 ----
 <1> Configure the settings for the cron job. See "Creating cron jobs" for more information on cron job settings.
 <2> The schedule for the job specified in link:https://en.wikipedia.org/wiki/Cron[cron format]. This example cron job runs every 30 minutes. Adjust the frequency as necessary, making sure to take into account how long the sync takes to run.
-<3> The LDAP sync command for the cron job to run. Passes in the sync configuration file that was defined in the config map.
-<4> This secret was created when the LDAP IDP was configured.
-<5> This config map was created when the LDAP IDP was configured.
+<3> How long to keep finished jobs. This should match the period of the job schedule to clean old failed jobs and prevent unnecessary alerts. See link:https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished[Kubernetes - TTL-after-finished Controller].
```
Comment (bergerhoffer) [inline modules/ldap-auto-syncing.adoc:213]:
```suggestion
<3> How long, in seconds, to keep finished jobs. This should match the period of the job schedule in order to clean old failed jobs and prevent unnecessary alerts. For more information, see link:https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished[TTL-after-finished Controller] in the Kubernetes documentation.
```

Just a few minor wording suggestions if you wouldn't mind updating. (Please make the changes manually and then squash - don't use the "Commit suggestion" button shown in the GitHub UI here)

Comment (sochotnicky) [reply modules/ldap-auto-syncing.adoc]: Thank you for the suggestion, text tweaked

**Comment by xingxingxia** [comment]
Sorry for late checking.
/lgtm

**Comment by bergerhoffer** [comment]
Thanks @sochotnicky for the update! The changes should be live on docs.openshift.com within an hour.

**Pull request merged**